gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -385,6 +400,42 @@ public void remove(Spec spec, Properties headers) throws IOException {
     }
   }
 
+  @Subscribe
+  public void handleKillFlowEvent(KillFlowEvent killFlowEvent) {
```

Comment:
I am a little concerned with the extent of hacks we are introducing because of the dag manager mode being present or not. I think we should evaluate doing this against just making DagManager the standard de-facto code path.